roles/sft-server/tasks/test.yml

```diff
@@ -20,3 +20,12 @@
   register: result
   until: result.stdout == "active"
   changed_when: false
+
+- name: "verifying that instance metrics are being exposed"
+  uri:
+    url: "https://127.0.0.1:{{ sft_nginx_metrics_port }}/metrics/node-exporter"
+    validate_certs: no
+    timeout: 6
+  register: response
+  failed_when: response.status != 400
+  when: sft_metrics_enabled | bool
```

Comment:
This doesn't work:

```
    TASK [sft-server : configuring SFT nginx vhost] ********************************
    changed: [molecule-hetznercloud-default-5395]
    
    TASK [sft-server : configuring metrics nginx vhost] ****************************
    changed: [molecule-hetznercloud-default-5395]
    
    TASK [sft-server : delaying test run to allow services to stabilize] ***********
    Pausing for 5 seconds
    (ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
    ok: [molecule-hetznercloud-default-5395]
    
    TASK [sft-server : verifying that service 'sft-server' started successfully] ***
    ok: [molecule-hetznercloud-default-5395]
    
    TASK [sft-server : verifying that service 'sft-server' is available] ***********
    ok: [molecule-hetznercloud-default-5395]
    
    TASK [sft-server : verifying that service 'nginx' runs successfully] ***********
    ok: [molecule-hetznercloud-default-5395]
    
    TASK [sft-server : verifying that instance metrics are being exposed] **********
fatal: [molecule-hetznercloud-default-5395]: FAILED! => {"changed": false, "content": "", "elapsed": 0, "failed_when_result": true, "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [Errno 111] Connection refused>", "redirected": false, "status": -1, "url": "https://127.0.0.1:8443/metrics/node-exporter"}
    
    RUNNING HANDLER [cloudalchemy.node-exporter : restart node_exporter] ***********
    
    RUNNING HANDLER [sft-server : reload systemd] **********************************
    
    RUNNING HANDLER [sft-server : reload nginx] ************************************
    
    PLAY RECAP *********************************************************************
    localhost                  : ok=17   changed=11   unreachable=0    failed=0    skipped=0    rescued=0    ignored=4   
    molecule-hetznercloud-default-5395 : ok=62   changed=43   unreachable=0    failed=1    skipped=19   rescued=0    ignored=0
    
ERROR:
```

I speculate that there's an ordering problem here: the reload nginx task/handler needs to run before this check. By including this test here as part of the playbook (as opposed to just being part of the molecule tests) there's an incorrect order.

You can either use 

```
- meta: flush_handlers
```

After writing to the nginx config; or detach the test.yml from the main run (so it runs instead only during molecule test).

Reply:
You are right, the pending nginx reload is breaking the test here. I adjusted the test so that it requests on the node exporter port directly and sidesteps nginx.